File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto/TouristAttraction.scala
package com.xebia.functional.xef.scala.conversation

import com.xebia.functional.xef.scala.conversation.*
import io.circe.Decoder
import com.xebia.functional.xef.prompt.Prompt

private final case class TouristAttraction(name: String, location: String, history: String) derives SerialDescriptor, Decoder

@main def runTouristAttraction: Unit = conversation {
  val statueOfLiberty: TouristAttraction = prompt(Prompt("Statue of Liberty location and history."))
  println(
    s"""
       |${statueOfLiberty.name} is located in ${statueOfLiberty.location} and has the following history:
       |${statueOfLiberty.history}
      """.stripMargin
  )
}

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/TouristAttraction.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.auto.llm.openai.conversation
import com.xebia.functional.xef.auto.llm.openai.prompt
import kotlinx.serialization.Serializable

@Serializable
data class TouristAttraction(val name: String, val location: String, val history: String)

suspend fun main() = conversation {
  val statueOfLiberty: TouristAttraction = prompt("Statue of Liberty location and history.")
  println(
    """|${statueOfLiberty.name} is located in ${statueOfLiberty.location} and has the following history:
                 |${statueOfLiberty.history}"""
      .trimMargin()
  )
}
